### PR TITLE
[2021.2] Fixing a shadow artefact on Terrain Detail objects

### PR DESCRIPTION
**Pull request by ellioman**
# Purpose of this PR
This PR fixes an issue with Shadow Artefacts appearing between cascades on Terrain Detail objects.

**This video shows the Before/After difference** 
https://user-images.githubusercontent.com/225349/117161557-4bafb880-adc2-11eb-96ac-e3f806524b39.mov

# What has been tested
Tested manually in the editor

# Comments to reviewers
This is the same fix done in https://github.com/Unity-Technologies/Graphics/pull/112 except now in TerrainDetailLitPasses.hlsl.

### CODE REVIEW

**Review by erikabar — APPROVED** [review]
LGTM

**Review by phi-lira — APPROVED** [review]

**Review by CertifiedRice — APPROVED** [review]

File: com.unity.render-pipelines.universal/Shaders/Terrain/TerrainDetailLitPasses.hlsl [hunk 1]
```diff
@@ -39,19 +37,23 @@ void InitializeInputData(Varyings input, out InputData inputData)
     inputData.positionCS = input.PositionCS;
     inputData.normalWS = half3(0, 1, 0);
     inputData.viewDirectionWS = half3(0, 0, 1);
-    #if defined(MAIN_LIGHT_CALCULATE_SHADOWS)
-    inputData.shadowCoord = input.ShadowCoords;
+
+    #if defined(REQUIRES_VERTEX_SHADOW_COORD_INTERPOLATOR)
+        inputData.shadowCoord = IN.shadowCoord;
```
Comment (nigeljw-unity) [inline com.unity.render-pipelines.universal/Shaders/Terrain/TerrainDetailLitPasses.hlsl:42]: This should be:
inputData.shadowCoord = input.shadowCoord;

Comment (ellioman) [reply com.unity.render-pipelines.universal/Shaders/Terrain/TerrainDetailLitPasses.hlsl]: Good catch!

**Comment by ellioman** [comment]
The yamato failure is known and already present on Master

**Comment by nigeljw-unity** [comment]
LGTM

**Pull request merged**